### DISCUSSION TITLE
SSL_set_tlsext_host_name not found in libssl-3-x64.dll. Confirmed by Dependency Walker

### DISCUSSION THREAD

**Discussion by OneBobone** [openssl/openssl#24183] [Q&A] [answered]
I believe that I need to set the SNI.

However, SSL_set_tlsext_host_name is not found in libssl-3-x64.dll. Have double checked by Dependency Walker. 
Version 3.3 dev from https://slproweb.com/products/Win32OpenSSL.html

The documentation says that it should be present!
What are my options?

Running OpenSSL on Win7 with tls1.2 and new ciphers. 
Trying to access a site hosted by aws cloudfront.
Commandline OpenSSL is successful. Have compared wireshark traffic and discovered minimum differences,

My code can access other sites with tls1.2 and new ciphers.

**Comment by mattcaswell** [comment, marked as answer]
`SSL_set_tlsext_host_name` is a macro so you would not expect to find it in the dll. It's defined here:

https://github.com/openssl/openssl/blob/e1fd043ad7fa865a8ef9160c892b49a098d23c71/include/openssl/tls1.h#L289-L291

As long as you include the `openssl/ssl.h` header file (which itself includes the `openssl/tls1.h` file above) then you should be able to use it. The underlying function called in the dll is `SSL_ctrl`

**Comment by OneBobone** [comment]
Thank you very much for that clarification!
My code now works. :-)

Maybe a good idea to mark in the documentation which functions are macros?
And their definition.

Cheers!
